lib/unison-sqlite/src/Unison/Sqlite/Connection.hs

```diff
@@ -511,9 +512,20 @@ rowsModified (Connection _ _ conn) =
 -- Vacuum
 
 -- | @VACUUM@
+--
+-- This action automatically retries on @SQLITE_BUSY@, which is returned when there is an open transaction on the
+-- database.
 vacuum :: Connection -> IO ()
 vacuum conn =
-  execute_ conn "VACUUM"
+  loop
+  where
+    loop =
```

Comment:
I'm generally pretty hesitant to add possibly infinite loops to the codebase; since vacuuming is always optional maybe we just try a couple times then give up? I dunno, maybe it's worse to "lie" about vacuuming.

In reality the argument is probably moot since this is unlikely to happen, but if in the case where a transaction is left open it'll be hard to debug why UCM fails to pull and just hangs forever, I think an exception or just a skipped vacuum is better in that case.

Reply:
Hmm, I don't think it would be difficult to debug the situation you describe. If you turn on query logging, you'll see a VACUUM attempt every 100ms.

But yes, we could instead try some hard-coded number of times.